engine/src/main/java/io/zeebe/engine/processor/workflow/job/JobBatchActivateProcessor.java

```diff
@@ -240,19 +241,19 @@ private void rejectCommand(
   private void raiseIncidentJobTooLargeForMessageSize(
       long jobKey, final JobRecord job, final TypedStreamWriter streamWriter) {
 
-    final DirectBuffer jobErrorMessage = job.getErrorMessageBuffer();
-    DirectBuffer incidentErrorMessage =
+    final String messageSize = ByteValue.prettyPrint(maxRecordLength);
+
+    final DirectBuffer incidentMessage =
         wrapString(
             String.format(
-                "The job with key '%s' is too large for the configured message size.", jobKey));
-    if (jobErrorMessage.capacity() > 0) {
-      incidentErrorMessage = jobErrorMessage;
-    }
+                "The job with key '%s' can not be activated because it plus additional activation-related data together is larger than the configured message size (%s). "
+                    + "Try to reduce the size by reducing the number of fetched variables or modifying the variable values.",
```

Comment:
The message is very precise but also a bit long. Maybe, we can remove some parts that are not so interesting for the user/operator.

```suggestion
                "The job with key '%s' can not be activated because it is larger than the configured message size (%s). "
                    + "Try to reduce the size by reducing the number of fetched variables or modifying the variable values.",
```